pkg/pattern/drain/line_tokenizer.go

```diff
@@ -51,22 +51,22 @@ func (p *punctuationTokenizer) Tokenize(line string) ([]string, interface{}) {
 		included := char < 128 && p.includeDelimiters[char] != 0
 		if char == ' ' || included || unicode.IsPunct(char) {
 			if i > start {
-				tokens[nextTokenIdx] = line[start:i]
+				tokens = append(tokens, line[start:i])
 				nextTokenIdx++
```

Comment:
You can delete nextTokenIdx completely now you're using append since it was only used to resize the returned slice.
You can potentially do the same for the spaces slice (use append, delete nextSpaceIdx) too.